src/model/analysis.ts

```diff
@@ -23,6 +23,11 @@ import {ResolvedUrl} from './url';
 import {Warning} from './warning';
 
 
+export type Result<G, B> = {
```

Comment:
Cool type.

Reply:
It's inspired by the Rust type of the same name. I've got a lib I was poking at that adds all the methods on the Rust type to a Result class. That was probably overkill, but some of them would be quite useful, like a `getOrDie` method

Reply:
G and B are a little meaningless to me. I guess "good" and "bad"? I'd go for T/E or V/E

Reply:
Done